tests/chainer_tests/iterators_tests/test_multiprocess_iterator.py

```diff
@@ -374,4 +391,162 @@ def test_iterator_serialize_backward_compat(self):
         self.assertAlmostEqual(it.previous_epoch_detail, 4 / 6)
 
 
+class TestMultiprocessIteratorConcurrency(unittest.TestCase):
+
+    def test_finalize_not_deadlock(self):
+        dataset = numpy.ones((1000, 1000))
+        it = iterators.MultiprocessIterator(dataset, 10, n_processes=4)
+        for _ in range(10):
+            it.next()
+
+        timeout = 5.
+        t = threading.Thread(target=lambda: it.finalize())
+        t.daemon = True
+        t.start()
+        t.join(timeout)
```

Comment:
How about `t.join(5)`?

Reply:
Though this is not my code, I agree with you.